lms/djangoapps/instructor/tests/views/test_instructor_dashboard.py

```diff
@@ -252,3 +268,51 @@ def test_grade_cutoffs(self):
         """
         response = self.client.get(self.url)
         self.assertIn('D: 0.5, C: 0.57, B: 0.63, A: 0.75', response.content)
+
+    @patch('instructor.views.api.MAX_STUDENTS_PER_PAGE_GRADE_BOOK', 2)
+    def test_calculate_page_info(self):
+        page = calculate_page_info(offset=0, total_students=2)
+        self.assertEqual(page["offset"], 0)
+        self.assertEqual(page["page_num"], 1)
+        self.assertEqual(page["next_offset"], None)
+        self.assertEqual(page["previous_offset"], None)
+        self.assertEqual(page["total_pages"], 1)
+
+    @patch('instructor.views.api.render_to_response', intercept_renderer)
+    @patch('instructor.views.api.MAX_STUDENTS_PER_PAGE_GRADE_BOOK', 1)
+    def test_spoc_gradebook_pages(self):
+        student = UserFactory.create()
+        CourseEnrollmentFactory.create(user=student, course_id=self.course.id)
+
+        student2 = UserFactory.create()
+        CourseEnrollmentFactory.create(user=student2, course_id=self.course.id)
+
+        # create grades for self.student as if they'd submitted the ccx
+        for chapter in self.course.get_children():
```

Comment:
Isn't this the same creation code as in `lms/djangoapps/ccx/tests/test_views.py`'s `setup` method? Is there a way to refactor, maybe a shared base class or helper method, so that you don't have this code duplicated? That way changes to the implementation will be reflected in both places.